test_cottage_page.js

```diff
@@ -0,0 +1,155 @@
+//Travels to a supplied cottage page and runs a suite of tests
+
+var url = "https://neontabs.neontribe.org/latest/hr/";
+var base_url = "https://neontabs.neontribe.org";
+var link_count = 0;
+var tests = 0;
+
+var area_heading = {
+	label: "Suffolk Cottages",
+	selector: "li.dropdown-item.dropdown-suffolk",
+	child_selector: 'li.dropdown.suffolk'
+};
+
+var area = {
+	label: "New Suffolk Cottages",
+	selector: "li.dropdown.suffolk a[href='/latest/hr/new-holiday-cottages?reference=ASTON,SWALLO,SHORE,GRACE,LAMP,TEAL,THATCH,MILLC,MANOR,RUTLAN,DRIFT,BROOKE,PETREL']",
+}
+
+var tmp_cottage_to_check = null;
+
+/*
+* Utils
+**/
+function pickRandomFromArray(arr) {
+	var max = arr.length;
+	var min = 0;
+	return arr[~~( Math.random() * (max - min) + min)];
+}
+//--//
+
+casper.start(casper.cli.get("target"), function() {}).run(function() {
+	casper.test.begin("Property Page Testing", tests, function suite(test) {
```

Comment:
This is a rather monolithic test. It would make sense to split it into multiple casper.test.begin chunks.

try a function to do the navigation to the property page, that you can then drop into the other tests to get you where you want to be.

it makes sense to test the shortlist functionality separately from the property page features.

Using the current structure, if you get a fail on introduction article, the test is likely to stop before the reviews tab test can be performed. We're usually interested in maximum diagnostic information per test run, so separating into another test would allow the suite to continue running in the event of a fail to get more information about what is and isn't working.